### PR TITLE
feat : Added watchlater

### PR DESCRIPTION
**Pull request by aiman-ansari**
- User can add videos to watchlater only if logged-in.
- User can add or remove videos to watchlater from homepage , videos page and single video page.

### CODE REVIEW

File: src/Pages/SingleVideo/SingleVideo.js [hunk 1]
```diff
@@ -1,11 +1,31 @@
...
+
 export const Singlevideo = () =>{
     const { videos} = useVideo()
     const { _id } = useParams()
+    const {addLike, likes, deleteLike} = useLikes()
+    const {addWatchLater, watchLater, deleteWatchLater} = useWatchLater()
+    const {isLogin} = useAuth()
+    const navigate = useNavigate()
+    console.log(likes)
+    console.log(watchLater)
```
Comment (rahulyadav139) [inline src/Pages/SingleVideo/SingleVideo.js]: Suggestion: remove these log messages.

File: src/Pages/SingleVideo/SingleVideo.js [hunk 2]
```diff
@@ -1,11 +1,31 @@
-import { useParams } from 'react-router-dom'
+import { useParams , useNavigate} from 'react-router-dom'
 import { useVideo } from '../../Context/VideoContext'
 import './SingleVideo.css'
 import { MustWatch, Single } from '../../Components/MustWatch/MustWatch'
+import { useAuth } from '../../Context/AuthContext'
+import { useLikes } from '../../Context/LikeContext'
```
Comment (rahulyadav139) [inline src/Pages/SingleVideo/SingleVideo.js]: suggestion: you can use barrel exporting and importing.

**Review by rahulyadav139 — APPROVED** [review]
Appreciation: The codes are readable. Great work.

**Review by TusharJade — APPROVED** [review]
UI is great, functionality is also working. Good to go for merge 👍

**Pull request merged**